### PR TITLE
Change default `max_shard_size` to smaller value

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

As per title, we can also change that value dynamically with respect to model size - with this change large models will end up having many shards.

cc @ArthurZucker

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -1896,9 +1896,11 @@ def save_pretrained(
                 Whether or not to push your model to the Hugging Face model hub after saving it. You can specify the
                 repository you want to push to with `repo_id` (will default to the name of `save_directory` in your
                 namespace).
-            max_shard_size (`int` or `str`, *optional*, defaults to `"10GB"`):
+            max_shard_size (`int` or `str`, *optional*, defaults to `"2GB"`):
                 The maximum size for a checkpoint before being sharded. Checkpoints shard will then be each of size
                 lower than this size. If expressed as a string, needs to be digits followed by a unit (like `"5MB"`).
+                We default it to 2GB in order for models to be able to run easily on free-tier google colab instances
+                without CPU OOM issues.
```
Comment (julien-c) [inline src/transformers/modeling_utils.py:1903]: maybe 5GB would be a better trade off no?

Comment (younesbelkada) [reply src/transformers/modeling_utils.py:1903]: Yeah makes sense

**Review by ArthurZucker — APPROVED** [review]
Thanks!

**Review by ArthurZucker — APPROVED** [review]
Good catch, we can't enforce the default is the same as from pretrained but no big deal

**Pull request merged**